src/main/java/aztech/modern_industrialization/machines/components/OverclockComponent.java

```diff
@@ -0,0 +1,189 @@
+    }
+
+    public InteractionResult onUse(MachineBlockEntity be, Player player, InteractionHand hand) {
+        ItemStack stackInHand = player.getItemInHand(hand);
+        var resourceInHand = Registry.ITEM.getKey(stackInHand.getItem());
+
+        for (Catalyst catalyst : catalysts) {
+            if (resourceInHand.equals(catalyst.resourceLocation) && stackInHand.getCount() >= 1) {
+                if (!player.isCreative()) {
+                    stackInHand.shrink(1);
+                }
+
+                if (TickMap.containsKey(catalyst.multiplier)) {
+                    var overclockTicks = TickMap.get(catalyst.multiplier);
+                    overclockTicks.value += catalyst.ticks;
+                } else {
+                    TickMap.put(catalyst.multiplier, new MutableTickCount(catalyst.ticks));
+                }
+
+                be.setChanged();
+                if (!be.getLevel().isClientSide()) {
+                    be.sync();
+                }
+                return InteractionResult.sidedSuccess(be.getLevel().isClientSide);
+            }
+        }
+
+        return InteractionResult.PASS;
+    }
+
+    public void tick(MachineBlockEntity be) {
+        var lastEntry = TickMap.lastEntry();
+        if (lastEntry != null) {
+            var overclockTicks = lastEntry.getValue();
+            overclockTicks.value--;
+            if (overclockTicks.value <= 0) {
+                TickMap.remove(lastEntry.getKey());
+            } else {
+                if (be.getLevel().isClientSide()) {
+                    for (int iter = 0; iter < 3; iter++) {
+                        var random = be.getLevel().getRandom();
+                        double d = be.getBlockPos().getX() + 0.5D;
+                        double e = be.getBlockPos().getY();
+                        double f = be.getBlockPos().getZ() + 0.5D;
+                        double i = random.nextDouble() * 0.6D - 0.3D;
+                        double k = random.nextDouble() * 0.6D - 0.3D;
+                        be.getLevel().addParticle(ParticleTypes.SMOKE, d + i, e + 1.05, f + k, 0.15 * (random.nextDouble() - 0.5), 0.15D,
+                                0.15 * (random.nextDouble() - 0.5));
+                    }
+                }
+            }
+        }
+    }
+
+    public List<Component> getTooltips() {
+        var tooltips = new ArrayList<Component>();
+        for (Catalyst catalyst : catalysts) {
+            Component catalystName;
+            if (Registry.BLOCK.containsKey(catalyst.resourceLocation)) {
+                catalystName = Component
+                        .translatable("block.%s.%s".formatted(catalyst.resourceLocation.getNamespace(), catalyst.resourceLocation.getPath()));
+            } else if (Registry.ITEM.containsKey(catalyst.resourceLocation)) {
+                catalystName = Component
+                        .translatable("item.%s.%s".formatted(catalyst.resourceLocation.getNamespace(), catalyst.resourceLocation.getPath()));
```

Comment:
You should get the `item` and call `item.getName` which will handle these cases correctly (and will also take into account mods overriding it).

Reply:
It looks slightly different with the brackets but Im guessing that's intended?
![image](https://user-images.githubusercontent.com/19756038/231944008-76ea1b86-3ebb-4dcc-9d09-1e175b4fa960.png)

Reply:
I believe you need `getHoverText`?

Reply:
![image](https://user-images.githubusercontent.com/19756038/232182581-4163c810-5b5e-4bf8-bf77-e8f904e31bc7.png)